grassroot-integration/src/main/java/za/org/grassroot/integration/location/LocationInfoBrokerImpl.java

```diff
@@ -367,6 +387,172 @@ public Province getProvinceFromGeoLocation(GeoLocation location) {
+
+        if(result.getBody() != null){
+            municipality = result.getBody().entrySet().iterator().next().getValue();
+
+            cache.put(new Element(userUid,municipality));
+        }
+
+        log.info("Municipality = {}",municipality);
+
+        return municipality;
+    }
+//    Loading users with location not null
+    @Override
+    public  void cacheMunicipalitiesForUsersWithLocation(){
+        List<UserLocationLog> userLocationLogs = userLocationLogRepository.findAll();
+
+        Map<String,Municipality> userMunicipalityMap = new HashMap<>();
+
+        for(UserLocationLog userLocationLog:userLocationLogs){
+            Municipality municipality =
+                            cacheMunicipalityByCoordinates(userLocationLog.getUserUid(),userLocationLog.getLocation().getLongitude(),userLocationLog.getLocation().getLatitude());
+            userMunicipalityMap.put(userLocationLog.getUserUid(),municipality);
+        }
+    }
+
+    @Override
+    public Map<String,Municipality> getMunicipalitiesForUsersWithLocationFromCache(Set<String> userUids){
+        Cache cache = cacheManager.getCache("user_municipality");
+        Map<String,Municipality> municipalityMap = new HashMap<>();
+
+        List<String> cacheKeys = cache.getKeys();
+
+        for(String userUid: userUids){
+            if(cacheKeys.contains(userUid)){
+                Municipality municipality = (Municipality) cache.get(userUid).getObjectValue();
+                municipalityMap.put(userUid,municipality);
+            }
+        }
+
+        log.info("Municipalities for users with location from cache is = {}",municipalityMap);
+        return municipalityMap;
+    }
+    //Counting the user location log within a period of a year
+    @Override
+    public int countUserLocationLogs(boolean countAll){
+
+        ConfigVariable configVariable = configRepository.findOneByKey("days.location.log.check").isPresent() ?
+                configRepository.findOneByKey("days.location.log.check").get() : null;
+
+        int userLocationLogsPeriod = 0;
+        if (configVariable != null) {
+            userLocationLogsPeriod = Integer.parseInt(configVariable.getValue());
+        }
+
+        Instant timeDaysAgo;
+
+        if(countAll){
+           //All Users count with the gps coordinates since the earliest instant , which is from the beginning of grassroots
+            timeDaysAgo = DateTimeUtil.getEarliestInstant();
+        }else{
+            //users count with gps coordinates within a period of a year
+            timeDaysAgo = Instant.now().minus(userLocationLogsPeriod, ChronoUnit.DAYS);
+        }
+        return userLocationLogRepository.countByTimestampGreaterThan(timeDaysAgo);
+    }
+
+    @Override
+    public void saveLocationLogsFromAddress(){
+        List<Address> addresses = addressRepository.loadAddressesWithLocation();
+        if(addresses != null){
+            for(Address address:addresses){
```

Comment:
Please start using Java 8 streams.
Also this is going to be highly inefficient. If there are 1000+ addresses (which is likely in current DB), then this is 1,000 queries. Rather use a bulk write.